scala/src/com/google/idea/blaze/scala/run/producers/BlazeScalaTestRunLineMarkerContributor.java

```diff
@@ -86,8 +100,16 @@ private static Info getInfo(ScClass testClass, @Nullable PsiElement testCase, In
       }
       url = "java:test://" + testClass.getQualifiedName() + "." + method.getName();
     } else if (testCase instanceof ScInfixExpr) {
-      // TODO: handle this case.
-      return null;
+      if (!TestNodeProvider.isSpecs2Expr(testCase)) {
+        return null;
+      }
+      Tuple2<ScTypeDefinition, String> pair = TestConfigurationUtil.specs2ConfigurationProducer()
+              .getLocationClassAndTest(new PsiLocation<>(testCase));
+      if (pair._2 == null) {
```

Comment:
This name doesn't include the "should" expression.
I'm doing this to get it.
Not sure if there's a cleaner way provided by the scala plugin.
```
      String testName = pair._2();
      if (testName == null) {
        return null;
      }
      ScInfixExpr testSuite = PsiTreeUtil.getParentOfType(testCase, ScInfixExpr.class);
      if (testSuite != null) {
        String should = testSuite.operation().refName();
        if (should.equals("should")) {
          testName =
              TestConfigurationUtil.getStaticTestName(testSuite.lOp(), false)
                  + should
                  + SmRunnerUtils.TEST_NAME_PARTS_SPLITTER
                  + testName;
        }
      }
      url = "java:test://" + testClass.getQualifiedName() + "." + testName;

```

Reply:
Yeah, I ended up doing something similar (found it in the scala plugin as well), see:
https://github.com/hmemcpy/intellij/blob/88f4debcf49bd5f4e9f50d0605f12468ef9a4036/scala/src/com/google/idea/blaze/scala/run/producers/BlazeSpecs2ConfigurationProducer.java#L135-L148

Maybe I could expose this as a utility method, seems useful.

To your question about the infix order, from what we use at Wix is mostly "should" block followed by an "in", but not always. Also valid are `>>` and, if you're using the immutable specs2 syntax, a `!` operator too. The call to `TestNodeProvider.isSpecs2Expr(testCase)` few lines above checks all that - it's a bit of code in the scala plugin that validates those infix expression order:
https://github.com/JetBrains/intellij-scala/blob/591723c14f1598b97eff720b20b0422f87795679/scala/scala-impl/src/org/jetbrains/plugins/scala/testingSupport/test/structureView/TestNodeProvider.scala#L335